test/tests/functional/pbs_runjob_hook.py

```diff
@@ -78,9 +83,33 @@ def test_normal_job_index(self):
         hook_name = "runjob_hook"
         attrs = {'event': "runjob"}
         rv = self.server.create_import_hook(hook_name, attrs,
-                                            self.hook_script, overwrite=True)
+                                            self.index_hook_script,
+                                            overwrite=True)
         self.assertTrue(rv)
         j1 = Job(TEST_USER)
         self.server.submit(j1)
         self.server.log_match("sub_job_array_index=None",
                               max_attempts=10, interval=1)
+
+    def test_reject_array_sub_job(self):
+        """
+        Test to check array subjobs,
+        jobs should run after runjob hook enabled set to false.
+        """
+        hook_name = "runjob_hook"
+        attrs = {'event': "runjob"}
+        rv = self.server.create_import_hook(hook_name, attrs,
+                                            self.reject_hook_script,
+                                            overwrite=True)
+        self.assertTrue(rv)
+        j1 = Job(TEST_USER)
+        lower = 1
+        upper = 3
+        j1.set_attributes({ATTR_J: '%d-%d' % (lower, upper)})
+        jid = self.server.submit(j1)
+        msg = "Not Running: PBS Error: Rejecting job by runjob hook event"
+        self.server.expect(JOB, {'job_state': 'Q', 'comment': msg}, id=jid)
+        a = {'enabled': 'false'}
+        self.server.manager(MGR_CMD_SET, HOOK, a, id=hook_name,
+                            expect=True, sudo=True)
+        self.server.expect(JOB, {'job_state': 'B'}, id=jid)
```

Comment:
don't you need to trigger a scheduling cycle here?

Reply:
Thanks @arungrover. Yes, i forgot to trigger the scheduling cycle but expect function was triggering itself that's why testcase got passed.
I'll add the needful change.